pkg/virt-launcher/virtwrap/agent-poller/agent_parser.go

```diff
@@ -194,16 +226,35 @@ func parseUsers(agentReply string) ([]api.User, error) {
 }
 
 // parseAgent gets the agent version from response
-func parseAgent(agentReply string) (string, error) {
-	result := AgentInfo{}
+func parseAgent(agentReply string) (GuestAgentVersion, error) {
+	gaInfo := AgentInfo{}
+	result := GuestAgentVersion{}
 	response := stripAgentResponse(agentReply)
 
-	err := json.Unmarshal([]byte(response), &result)
+	err := json.Unmarshal([]byte(response), &gaInfo)
 	if err != nil {
-		return "", err
+		return result, err
 	}
 
-	return result.Version, nil
+	unsupported := false
+	for _, cmd := range RequiredGuestAgentCommands {
+		found := false
+		for _, gaCmd := range gaInfo.SupportedCommands {
+			if (cmd == gaCmd.Name) && (gaCmd.Enabled) {
+				found = true
+				break
+			}
+		}
```

Comment:
Here's something i'm not 100% sure what to do about... the required guest agent commands are actually dynamic depending on the VMI spec. 

For example, the following are only required when accessCredentials are set on the VMI. Otherwise they are not required. 

```
	"guest-set-user-password",
	"guest-exec-status",
	"guest-exec",
	"guest-file-open",
	"guest-file-close",
	"guest-file-read",
	"guest-file-write",
```

Maybe the list of commands should be passed back to virt-handler and since virt-handler has knowledge of the VMI in question it can make the right decision

Reply:
Circling back to this.

I see what you're saying, but I see a way things could get weird. It's possible for virt-launcher to remain running after an update of KubeVirt. In such a case, the could exist a delta between what virt-handler thinks virt-launcher needs and what it's actually going to use. I'm not sure how big a deal this is.

I'd argue for using a static list for simplicity, but looking at the list you used as an example, I'm zeroing in on `guest-exec`. That API call is commonly disabled on guests because of its enormous power.

Reply:
"guest-exec-status",
	"guest-exec",
	"guest-file-open",
	"guest-file-close",
	"guest-file-read",
	"guest-file-write",

belong to a yaml containing accessCredentials with a sshPubKey using qemuGuestAgent propagation

```
  accessCredentials:
  - sshPublicKey:
      source:
        secret:
          secretName: my-pub-key 
      propagationMethod:
          qemuGuestAgent:
              users:
                - "fedora"
```

"guest-set-user-password",

belongs to a yaml with userPassword accessCredential using the qemuGuestAgent propagation method

```
  accessCredentials:
   - userPassword:
         source:
             secret:
                 secretName: my-pub-key2 
          propagationMethod:
              qemuGuestAgent: {}
```